pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -42,17 +42,22 @@ static __always_inline http2_stream_t *http2_fetch_stream(const http2_stream_key
     return bpf_map_lookup_elem(&http2_in_flight, http2_stream_key);
 }
 
-// Similar to read_var_int, but with a small optimization of getting the current character as input argument.
-static __always_inline bool read_var_int_with_given_current_char(struct __sk_buff *skb, skb_info_t *skb_info, __u64 current_char_as_number, __u64 max_number_for_bits, __u64 *out) {
+// Similar to read_hpack_int, but with a small optimization of getting the
+// current character as input argument.
+static __always_inline bool read_hpack_int_with_given_current_char(struct __sk_buff *skb, skb_info_t *skb_info, __u64 current_char_as_number, __u64 max_number_for_bits, __u64 *out) {
     current_char_as_number &= max_number_for_bits;
 
+    // In HPACK, if the number is too big to be stored in max_number_for_bits
+    // bits, then those bits are all set to one, and the rest of the number must
+    // be read from subsequent bytes.
     if (current_char_as_number < max_number_for_bits) {
         *out = current_char_as_number;
         return true;
     }
 
+    // Read the next byte, and check if it is the last byte of the number.
     __u64 next_char = 0;
-    if (bpf_skb_load_bytes(skb, skb_info->data_off, &next_char, 1) >=0 && (next_char & 128) == 0) {
+    if (bpf_skb_load_bytes(skb, skb_info->data_off, &next_char, 1) >= 0 && (next_char & 128) == 0) {
```

Comment:
You can add here - unlike the general algorithm, we support indexes up to X, due to ebpf-complexity limitations ....

(elaborate/rephrase)